src/code/z_camera.c

```diff
@@ -6919,9 +6919,15 @@ s32 Camera_Special9(Camera* camera) {
         case 0:
             camera->stateFlags &= ~(CAM_STATE_1 | CAM_STATE_2);
             camera->animState++;
-            rwData->targetYaw = ABS(playerPosRot->rot.y - adjustedPlayerPosRot.rot.y) >= 0x4000
-                                    ? adjustedPlayerPosRot.rot.y - 0x7FFF
-                                    : adjustedPlayerPosRot.rot.y;
+
+            //! @bug There should be a cast to s16 in the condition of the ternary below.
```

Comment:
so you're saying the result of `ABS()` should be cast to s16?

Reply:
the angle subtraction inside of the ABS. like usual subtraction on binary angles always needs to be casted

Reply:
I agree the wording doesn't make clear where the cast is missing.

I originally thought on something like this (casting the result of the operation)
```c
            rwData->targetYaw = (s16)(ABS(playerPosRot->rot.y - referencePosRot.rot.y) >= 0x4000
                                    ? referencePosRot.rot.y - 0x7FFF
                                    : referencePosRot.rot.y);
```

But the cast should be inside the ABS?
```c
            rwData->targetYaw = ABS((s16)(playerPosRot->rot.y - referencePosRot.rot.y)) >= 0x4000
                                    ? referencePosRot.rot.y - 0x7FFF
                                    : referencePosRot.rot.y;
```

Reply:
Yes, on the subtraction itself. This is just a fact of subtracting binary angles in general that its needed, not specific to this particular case. Which is why I didnt go into very much detail here

Reply:
The other subtraction (`referencePosRot.rot.y - 0x7FFF`) made me think the whole result needed to be casted. Idk, it may be just be me. Either way the pr seems fine

Reply:
that one gets casted implicitly on assignment to `targetYaw`, the one in the condition does not.
I can change the wording, but just making it clear that this is a general requirement and not something weird to this case

Reply:
```suggestion
            //! @bug The angle passed to ABS should be cast to s16.
```
?

Reply:
sure